### PR TITLE
Update settings to use pathlib instead of os.path

### PR DESCRIPTION
**Pull request by klavman**
Connects to https://github.com/archivematica/Issues/issues/1622

### CODE REVIEW

**Comment by replaceafill** [comment]
@klavman feel free to ping me here when this is ready for review. Also, please submit your [Contributor's Agreement](https://github.com/artefactual/archivematica/blob/qa/1.x/CONTRIBUTING.md#contributors-agreement) if you haven't yet.

**Comment by klavman** [comment]
@replaceafill I've thoroughly tested the changes, and everything appears to be working as expected. The pull request is ready for review.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -35,6 +35,22 @@ repos:
     additional_dependencies:
     - flake8-bugbear==23.9.16
     - flake8-comprehensions==3.14.0
+    exclude: |
+      (?x)^(
+        storage_service/storage_service/settings/.*\.py
+      )
+- repo: https://github.com/pycqa/flake8
```
Comment (replaceafill) [inline .pre-commit-config.yaml:42]:
```suggestion
- repo: https://github.com/pycqa/flake8
  # This is a temporary hook to port os.path code to pathlib gradually. Add
  # the ported modules/packages explicitly to `files` below. For reference see
  # https://github.com/archivematica/Issues/issues/1622#issuecomment-1912524642
```

**Review by replaceafill — APPROVED** [review]
This is a great start @klavman! Thank you. I suggested a comment for the temporary hook.

**Pull request merged**